swap/src/monero/wallet_rpc.rs

```diff
@@ -163,11 +117,116 @@ impl WalletRpc {
         })
     }
 
-    fn tar_path(&self) -> PathBuf {
+    fn archive_path(&self) -> PathBuf {
         self.working_dir.join("monero-cli-wallet.tar")
     }
 
     fn exec_path(&self) -> PathBuf {
         self.working_dir.join(PACKED_FILE)
     }
+
+    async fn download(monero_wallet_rpc: &Self) -> Result<()> {
+        use anyhow::{bail, Context};
+        use async_compression::tokio::bufread::BzDecoder;
+        use tokio_tar::Archive;
+
+        let mut options = OpenOptions::new();
+        let mut file = options
+            .read(true)
+            .write(true)
+            .create_new(true)
+            .open(monero_wallet_rpc.archive_path())
+            .await?;
+
+        let byte_stream = reqwest::get(DOWNLOAD_URL)
+            .await?
+            .bytes_stream()
+            .map_err(|err| std::io::Error::new(ErrorKind::Other, err));
+
+        let mut stream = FramedRead::new(
+            BzDecoder::new(StreamReader::new(byte_stream)),
```

Comment:
This line is actually the only difference between the download of linux/macos and windows. I think we could make the part that is OS-specific a lot smaller here.

Happy to leave that to a follow-up though.

Reply:
Agreed! most of the `download` is duplicated. Follow-up OK for me as well.

Reply:
fixed it